jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1303,11 +1315,24 @@ public Field getField(String name) throws NoSuchFieldException, SecurityExceptio
  * @return		Class<?>[]
  *					the interfaces the receiver claims to implement.
  */
-public Class<?>[] getInterfaces()
-{
-	return J9VMInternals.getInterfaces(this);
+public Class<?>[] getInterfaces(){
+	if (cachedInterfaces == null){
+		if (cachedInterfacesOffset == -1) {
+			Field cachedInterfacesField;
+			try {
+				cachedInterfacesField = Class.class.getDeclaredField("cachedInterfaces"); //$NON-NLS-1$
+				cachedInterfacesOffset = getUnsafe().objectFieldOffset(cachedInterfacesField);
+			} catch (NoSuchFieldException e) {
+				throw newInternalError(e);
+			}
+		}
+		getUnsafe().putOrderedObject(this, cachedInterfacesOffset, J9VMInternals.getInterfaces(this));
+	}
+	Class<?>[] newInterfaces = cachedInterfaces ? cachedInterfaces.clone(): null;
```

Comment:
If `cachedInterfaces` can't be `null`, this check is redundant.

Reply:
A possible optimization - if `cachedInterfaces.length == 0`, we can skip the clone and return the original as it can't be modified.